pkg/db/query/job_queries.go

```diff
@@ -12,6 +12,56 @@ import (
 	"github.com/openshift/sippy/pkg/filter"
 )
 
+func JobRunTestCount(dbc *db.DB, jobRunID int64) (int, error) {
+	var prowJobRunTestCount int
+	var tests []models.ProwJobRunTest
+
+	res := dbc.DB.Find(&tests, "prow_job_run_id = ?", jobRunID)
+
+	if res.Error != nil {
+		return -1, res.Error
+	}
+
+	prowJobRunTestCount = len(tests)
+
+	return prowJobRunTestCount, nil
+}
+
+func ProwJobHistoricalTestCounts(dbc *db.DB, prowJobID uint) (int, error) {
+
+	var jobRuns []models.ProwJobRun
+	res := dbc.DB.Joins("ProwJob").
+		Preload("Tests").Order("updated_at desc").Limit(10).Find(&jobRuns, "prow_job_id = ? AND succeeded = true", prowJobID)
+	if res.Error != nil {
+		return 0, res.Error
+	}
+
+	if len(jobRuns) == 0 {
+		// select * from prow_job_runs where prow_job_id = 1805 and succeeded = true limit 10;
+		// has no 'succeeded = true' runs
+		// if we didn't get any then open it up to just the most recent runs
+		res = dbc.DB.Joins("ProwJob").
+			Preload("Tests").Order("updated_at desc").Limit(10).Find(&jobRuns, "prow_job_id = ?", prowJobID)
+		if res.Error != nil {
+			// we didn't get an error on our first query so we can use that result (0) and continue
+			log.Warnf("Error during fallback historical count query for prowJobId %d", prowJobID)
+		}
+	}
+
+	t := 0
+
+	if len(jobRuns) > 0 {
+
+		for _, jobRun := range jobRuns {
+			t += len(jobRun.Tests)
+		}
+
+		t /= len(jobRuns)
```

Comment:
/= ? TIL!

Reply:
heh, I only learned since I got  a lint nag

Reply:
Getting the avg number of tests would be faster to do in SQL

Example:

```sql
SELECT
    avg(count)
FROM (
    SELECT
        prow_job_run_id,
        count(*)
    FROM
        prow_job_run_tests
        INNER JOIN prow_job_runs ON prow_job_runs.id = prow_job_run_tests.prow_job_run_id
        INNER JOIN prow_jobs ON prow_jobs.id = prow_job_runs.prow_job_id
    WHERE
        prow_jobs.name = 'periodic-ci-openshift-release-master-ci-4.13-e2e-aws-ovn-upgrade'
        AND prow_job_runs.timestamp >= CURRENT_DATE - interval '7' day
    GROUP BY
        prow_job_run_id) t;
```

Reply:
All that new knowledge lost with more efficient sql queries...  I was wondering about raw query performance vs gorm.  Thanks for the pointer.